Sourcery/Parsing/FileParser.swift

```diff
@@ -573,33 +578,46 @@ extension FileParser {
         case ("="?, _?):
-             let body = wrappedBody.substring(from: wrappedBody.index(after: wrappedBody.startIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
-             rawValue = parseEnumValues(body)
+            let body = wrappedBody.substring(from: wrappedBody.index(after: wrappedBody.startIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
+            rawValue = parseEnumValues(body)
         case ("("?, ")"?):
-             let body = wrappedBody.substring(with: wrappedBody.index(after: wrappedBody.startIndex)..<wrappedBody.index(before: wrappedBody.endIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
-             associatedValues = parseEnumAssociatedValues(body)
+            let body = wrappedBody.substring(with: wrappedBody.index(after: wrappedBody.startIndex)..<wrappedBody.index(before: wrappedBody.endIndex)).trimmingCharacters(in: .whitespacesAndNewlines)
+            associatedValues = parseEnumAssociatedValues(body)
         case (nil, nil):
             break
         default:
-             Log.warning("\(logPrefix)parseEnumCase: Unknown enum case body format \(wrappedBody)")
+            Log.warning("\(logPrefix)parseEnumCase: Unknown enum case body format \(wrappedBody)")
         }
 
+        // Parse enum case annotations
+
         let annotations: [String: NSObject]
-        if var body = extract(.keyPrefix, from: source)?.trimmingSuffix("case") {
-            // search backwards for possible enum cases separators
-            if let semicolon = body.range(of: ";", options: [.backwards])?.upperBound {
-                body = body.substring(from: semicolon)
-            } else if let comma = body.range(of: ",", options: [.backwards])?.upperBound {
-                body = body.substring(from: comma)
-            } else if let parenthesis = body.range(of: ")", options: [.backwards])?.upperBound {
-                body = body.substring(from: parenthesis)
-            } else if let `case` = body.range(of: "case", options: [.backwards])?.upperBound {
-                body = body.substring(from: `case`)
-            } else if let brace = body.range(of: "{", options: [.backwards])?.upperBound {
-                body = body.substring(from: brace)
-            }
 
-            annotations = AnnotationsParser(contents: body).all
+        var annotationRangeStart: Int64?
+        if let previousCase = `enum`.cases.last, let previousCaseKeyRange = Substring.key.range(for: previousCase.__underlyingSource) {
+            // if current enum case is not the first one in containing enum
+            // get everything after the previous case declaration
+            annotationRangeStart = previousCaseKeyRange.offset + previousCaseKeyRange.length
+        } else if let enumBody = Substring.body.range(for: `enum`.__underlyingSource) {
+            // if current enum case is the first one in containing enum
+            // get everything after enum body start
+            annotationRangeStart = enumBody.offset
```

Comment:
what if I had a variable in enum and added annotation to it, would you parse it as part of the first enum case? 

The rules for parsing annotations I had originally were that you go backwards from the current code element and stop immediately if there is a non comment line, they were only tweaked for the enum associated value annotations